llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -750,20 +755,29 @@ LazyValueInfoImpl::solveBlockValueNonLocal(Value *Val, BasicBlock *BB) {
       // Explore that input, then return here
       return std::nullopt;
 
-    Result.mergeIn(*EdgeResult);
-
-    // If we hit overdefined, exit early.  The BlockVals entry is already set
-    // to overdefined.
-    if (Result.isOverdefined()) {
-      LLVM_DEBUG(dbgs() << " compute BB '" << BB->getName()
-                        << "' - overdefined because of pred '"
-                        << Pred->getName() << "' (non local).\n");
-      return Result;
+    IncomingIVs.push_back(*EdgeResult);
+    if (EdgeResult->isOverdefined()) {
+      OverdefinedPred = Pred;
+      break;
     }
+
     if (PerPredRanges)
       PredLatticeElements->insert({Pred, *EdgeResult});
   }
 
+  Result.mergeIn(IncomingIVs);
+
+  // If we hit overdefined, exit early.  The BlockVals entry is already set
+  // to overdefined.
+  if (Result.isOverdefined()) {
+    LLVM_DEBUG(dbgs() << " compute BB '" << BB->getName()
+                      << "' - overdefined because of pred";
+               if (OverdefinedPred) 
+                 dbgs() << " '" << OverdefinedPred->getName() << "'";
```

Comment:
The clangd complains about the code format and suggests fixing it as follows:
```suggestion
               if (OverdefinedPred) dbgs()
               << " '" << OverdefinedPred->getName() << "'";
```
I believe it is an FP caused by code within the MACRO (`LLVM_DEBUG(... <code> ...)`.